addon/services/key-manager.js

```diff
@@ -79,88 +82,97 @@ export default Ember.Service.extend({
+      if (!modifierKeyCodes.includes(keyCode)) {
+        get(this, 'downKeys').addObject(keyCode);
+      }
+
+      return this._findComboByName(eventName);
+    } finally {
+      this._clearExecutionKeys(event);
     }
+  },
 
-    modifierKeys.forEach((key) => {
-      const prop = `${key}Key`;
-      set(this, prop, event[prop]);
-    });
+  handler(event) {
+    if (get(this, 'isDestroyed') || get(this, 'isDestroying')) { return; }
+
+    const combo = this.findMatchingCombo(event);
+    const runLoopGuard = !get(this, 'matchFound');
 
-    if (data) {
-      const { eventName } = data;
-      const combo = this._findComboByName(eventName);
-      const runLoopGuard = !get(this, 'matchFound');
-
-      if (combo && runLoopGuard) {
-        set(this, 'matchFound', true);
-        run.next(() => {
-          set(this, 'matchFound', false);
-        });
-
-        const isNotOnInput = inputElements.every(e => !$(document.activeElement).is(e));
-        if (!get(combo, 'disableOnInput') || isNotOnInput) {
-          const callback = get(combo, 'callback');
-          if (callback) {
-            callback(event);
-          }
+    if (combo && runLoopGuard) {
+      set(this, 'matchFound', true);
+      run.next(() => {
+        set(this, 'matchFound', false);
+      });
+
+      const isNotOnInput = inputElements.every(e => !$(document.activeElement).is(e));
+      if (!get(combo, 'disableOnInput') || isNotOnInput) {
+        const callback = get(combo, 'callback');
+        if (callback) {
+          callback(event);
         }
       }
     }
+  },
 
+  executionKeys: Ember.computed(function() {
+    return Object.keys(keyCodes).map((key) => {
+      return keyCodes[key];
+    }).reject((code) => {
+      return modifierKeyCodes.includes(code);
+    });
+  }),
+
+  _clearExecutionKeys(event) {
     if (event.type === 'keyup') {
       get(this, 'downKeys').removeObject(event.keyCode);
     }
 
-    this._clearExecutionKeys();
-  },
-
-  _clearExecutionKeys() {
-    const executionKeys = Object.keys(keyCodes).map((key) => {
-        return keyCodes[key];
-      })
-      .reject((code) => {
-        return modifierKeyCodes.includes(code);
-      });
-    get(this, 'downKeys').removeObjects(executionKeys);
+    get(this, 'downKeys').removeObjects(this.get('executionKeys'));
   },
 
   _findComboByName(eventName) {
     const combos = get(this, 'combos');
-    const combosWithKeys = this._combosWithKeys(combos)
-      .sortBy('priority');
-    const comboWithName = combosWithKeys.findBy('eventName', eventName);
-
-    if (!comboWithName) {
-      return false;
-    }
-
-    const highestPriority = get(combosWithKeys, 'lastObject.priority');
-    const comboWithNamePriority = get(comboWithName, 'priority');
-
-    if (comboWithNamePriority >= highestPriority) {
-      return comboWithName;
+    const combosWithNameAndKeys = combos
+          .filterBy('eventName', eventName)
+          .filter((combo) => {
```

Comment:
And check that the combo has matching keys
ie: Make a function that checks for matching keys and returns a boolean